Use correct bolus values in CBF calculation and temporarily disable Q2TIPS, multi-PLD, and no-BolusCutOff processing

tsalo opened the pull request:
Closes #167.

## Changes proposed in this pull request
- For (P)CASL data, use `LabelingDuration` as the bolus value in BASIL CBF calculation. For PASL data, use `BolusCutOffDelayTime` instead.
- Apply the same fix to the vanilla CBF calculation.
- Raise an error if PASL data do not have BolusCutOffFlag set to True.
- Hopefully support QUIPSS PASL data as well.
- Explicitly block multi-PLD and Q2TIPS data for now.

I still need to figure out what to do if BolusCutOffFlag is False, as well as if BolusCutOffTechnique is QUIPSS or Q2TIPS.

tsalo committed:
Use BolusCutOffDelayTime for PASL data.

tsalo committed:
Add fix to vanilla CBF calculation.

tsalo committed:
Use nilearn for masking and unmasking.

tsalo committed:
Update misc.py

tsalo committed:
Update misc.py

tsalo committed:
Merge remote-tracking branch 'upstream/main' into pasl-bug

tsalo committed:
Move compute_cbf into ComputeCBF.

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Remove logger messages.

tsalo committed:
Require BCOF be True for PASL across CBF methods.

tsalo committed:
Remove unused att output.

tsalo committed:
Remove unused BASIL inputs.

Not sure why the output filenames can be set as inputs when those inputs are never used...

tsalo committed:
Remove more inputs that are really outputs.

tsalo committed:
Update ge.py

andrewrosss commented:
I'm not a core team member, but I'm the one who opened #167. These changes seem reasonable to me, specifically, using `LabelingDuration` for (P)CASL and `BolusCutOffDelayTime` for PASL seems to adhere to the intent of the BIDS spec and would fix address the original issue.

andrewrosss commented on aslprep/utils/misc.py:
```diff
@@ -232,13 +232,22 @@ def _prefix(subid):
-    return pcasl1
+    aslt = metadata["ArterialSpinLabelingType"]
+
+    if "CASL" in aslt:
```
nit: I think it would be better to have this be `if aslt in ["CASL", "PCASL"]:`

tsalo replied:
That's a good point. Thanks!

tsalo committed:
Check ASLT more explicitly.

tsalo committed:
Expect a list if technique is Q2TIPS.

tsalo commented:
Thank you for reviewing @andrewrosss. Your input is very much appreciated.

andrewrosss commented:
Happy to help! :) Thanks for the fix!

tsalo committed:
Remove duplicate references.

tsalo committed:
Work around nilearn-made singleton dims.

Nilearn's maskers add a singleton time dimension when the data are purely spatial.

tsalo committed:
Pin nilearn version.

Starting in 0.12.0, nilearn masker.transform on a 3D image will return a 1D array.

tsalo committed:
Improve comment.

tsalo committed:
Clean up CBF calculation.

tsalo committed:
Fix PASL CSF calculation.

tsalo committed:
Improve comments.

tsalo committed:
Improve ComputeCBF input/output names.

tsalo committed:
Improve documentation again.

tsalo committed:
Remove unnecessary fields from BibTeX entries.

tsalo committed:
Document ComputeCBF.

tsalo committed:
Update cbf_computation.py

tsalo committed:
Improve variable naming.

tsalo committed:
Change a lot of variable names.

tsalo committed:
Hopefully support QUIPSS and Q2TIPS.

tsalo committed:
Update.

tsalo committed:
Add a message about Q2TIPS.

tsalo committed:
Merge remote-tracking branch 'upstream/main' into pasl-bug

tsalo committed:
Add a test for ComputeCBF.

tsalo committed:
Update test_interfaces_cbf_computation.py

tsalo committed:
Update test_interfaces_cbf_computation.py

tsalo committed:
Just use tmpdir.

tsalo committed:
Update test_interfaces_cbf_computation.py

tsalo committed:
Add CASL labeling efficiency value.

tsalo committed:
Replace T1blood formula with hardcoded values.

tsalo committed:
Drop check within pcasl_or_pasl. Checked elsewhere.

tsalo committed:
Push T1blood change to another PR.

tsalo committed:
Modify PLDs in metadata during deltam selection.

tsalo committed:
Try fixing the PLDs.

tsalo committed:
Add logger messages.

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Update cbf_computation.py

tsalo committed:
Fix?

tsalo committed:
Add another integration test.

tsalo committed:
Update test.

tsalo committed:
Skip new test for now.

tsalo committed:
Try to simplify the CBF workflows.

tsalo committed:
Fix bugs hopefully.

tsalo committed:
Move connections around a bit.

tsalo committed:
Fix bug maybe.

tsalo committed:
Expect failures on multi-PLD data.

tsalo committed:
Fix test.

tsalo committed:
Fix things more.

tsalo committed:
Update cbf_computation.py

tsalo committed:
Fix...

tsalo committed:
Update cbf_computation.py

tsalo committed:
Use first BolusCutOffDelayTime for Q2TIPS w BASIL.

tsalo committed:
Split things up a bit.

tsalo committed:
Disable Q2TIPS.
